src/app/shared/nav/nav-menu.component.spec.ts

```diff
@@ -73,7 +75,12 @@ describe('NavMenuComponent', () => {
       location = TestBed.get(Location);
       router.initialNavigation();
 
-      comp.store.dispatch(new CastleFilterAction({filter}));
+      // This action is normally sent out by the ngrx StoreRouterConnectingModule
+      // and _not_ by the application
+      // We're mocking it up here so our custom router reducer reducer runs and the router selector gives us the state
```

Comment:
fancy!

Reply:
Oh! This comment is why I actually used a routing effect to translate the action to a CUSTOM_ROUTER_NAVIGATION. That event nonsense I was talking about below was referring to trying to mock up a RoutesRecognized event that was in the original action. Annoying. Stripping it down to what the application was interested in seemed better anyway.

I will make some comment updates because this one is a bit stale.